### DISCUSSION TITLE
Google Provider in Production

### DISCUSSION THREAD

**Discussion by niall-byrne** [nextauthjs/next-auth#3027] [Help]
### Description 🐜

I'm seeing problems with the google provider in production (works fine on localhost with the same config).

### Is this a bug in your own project?

No

### How to reproduce ☕️

```
GoogleProvider({
  clientId: process.env.AUTH_GOOGLE_ID,
  clientSecret: process.env.AUTH_GOOGLE_SECRET,
}),
```

Scopes on google side are email and profile (I have experimented with the openID scope but it makes no difference.)
Works fine on localhost, does not work when deployed to Netlify.

Other oauth providers (Github, Spotify, Facebook all work fine in production - only Google is problematic.)

### Screenshots / Logs 📽

```
https://next-auth.js.org/errors#signin_oauth_error The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined {
  error: {
    message: 'The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined',
    stack: 'TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined\n' +
      '    at Function.from (buffer.js:330:9)\n' +
      '    at hkdf_extract (/var/task/node_modules/futoin-hkdf/hkdf.js:77:57)\n' +
      '    at hkdf (/var/task/node_modules/futoin-hkdf/hkdf.js:147:17)\n' +
      '    at hkdf (/var/task/node_modules/next-auth/jwt/index.js:147:32)\n' +
      '    at getDerivedEncryptionKey (/var/task/node_modules/next-auth/jwt/index.js:181:18)\n' +
      '    at Object.encode (/var/task/node_modules/next-auth/jwt/index.js:62:97)\n' +
      '    at createPKCE (/var/task/node_modules/next-auth/server/lib/oauth/pkce-handler.js:40:43)\n' +
      '    at getAuthorizationUrl (/var/task/node_modules/next-auth/server/lib/oauth/authorization-url.js:44:52)\n' +
      '    at processTicksAndRejections (internal/process/task_queues.js:97:5)\n' +
      '    at async Object.signin (/var/task/node_modules/next-auth/server/routes/signin.js:30:32)',
    name: 'TypeError'
  },
  provider: {
    id: 'google',
    name: 'Google',
    type: 'oauth',
    wellKnown: 'https://accounts.google.com/.well-known/openid-configuration',
    authorization: { params: [Object] },
    idToken: true,
    checks: [ 'pkce', 'state' ],
    profile: [Function: profile],
    clientId: <REDACTED>',
    clientSecret: '<REDACTED>',
    signinUrl: 'https://stage-mla.netlify.app/api/auth/signin/google',
    callbackUrl: 'https://stage-mla.netlify.app/api/auth/callback/google'
  },
  message: 'The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined'
}
```

### Environment 🖥

Deployed to Netlify.
Again works fine when running on localhost.
Other oauth providers (Github, Spotify, Facebook all work fine in production - only Google is problematic.)

  System:
    OS: macOS 10.15.7
    CPU: (4) x64 Intel(R) Core(TM) i3-8100B CPU @ 3.60GHz
    Memory: 1.50 GB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.18.1 - ~/.asdf/installs/nodejs/14.18.1/bin/node
    npm: 6.14.15 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 94.0.4606.81
  npmPackages:
    next: ^11.1.2 => 11.1.2
    next-auth: ^4.0.0-beta.4 => 4.0.0-beta.4
    react: ^17.0.2 => 17.0.2

### Contributing 🙌🏽

Yes,  I am willing to help solve this bug in a PR

**Comment by jessfraz** [comment]
I am also having problems with the google provider in v4, not sure if the same issue, but when using the prisma database adapter it saves the user but not the session or account

GitHub and e-mail auth or working fine. It's just google. And I've never had troubles before v4 and without a database.

**Comment by jessfraz** [comment]
I am also unsure how to debug this as there is no error just in the URL error=Callback and in the UI: `Try signing in with a different account.`
